### ISSUE TITLE
Save user application data in %LOCALAPPDATA% under Windows

### ISSUE THREAD

**Issue by kodmord** [OpenRCT2/OpenRCT2#10042]
**OS:** [e.g. Windows 10]
**Version:** [e.g. 0.2.3]
**Commit/Build:** [e.g. 15767c3]

Currently, all application data is stored in Documents\OpenRCT2. I think they should be stored in %LOCALAPPDATA%\OpenRCT2 (%USERPROFILE%\AppData\Local\OpenRCT2). What do you think? I can make a pull request.

**Comment by Gymnasiast** [comment]
As far as I know, %LOCALAPPDATA% is intended for stuff that has to stay on the machine - stuff like caches. %APPDATA% is meant for stuff that is not machine-specific and can thus safely be synced when you're using roaming profiles. (A distinction which gets important when you're running programs in domains.)

For stuff like caches and settings, though, you're right that they don't strictly belong in Documents. Colour schemes are also something that would normally go into %appdata%. You could also argue that Pictures is a better place for the screenshots (for which we already have a ticket) and that we could put Saved Games in the folder of the same name.

But for stuff like track designs, scenarios and objects, this is a lot less clear-cut - they are not config, but you could call them documents.

If you go down that route, though, you will end up with stuff all over the place. That makes backing up and  installing custom content more difficult. Two years ago, we had a similar discussion about the save locations on Linux (for which something very similar applies) and back then we decided not to go with it. (Though I won't rule it out.)

Now, I think this warrants discussion. I'm not speaking out in favour of any option on Windows (I think that is best left to people who use it as their _primary_ OS), but I figured I'd provide some context and some pros and cons.

**Comment by ZehMatt** [comment]
I would actually agree that some stuff should/could go in %APPLOCALDATA% but then we are not actually producing a lot of files so I think the current way is not bad either, perhaps you could share your reasoning on why you would like to change it?

**Comment by telk5093** [comment]
I think the current one is better to find the location for newbies.

**Comment by wisepotato** [comment]
Is this configurable at the moment? Why not allow for a custom place? A use could use OneDrive as their saving location for example (which would be great).

**Comment by ZehMatt** [comment]
You can already save/load your parks from wherever you like. Also you can already override the data directory via argument but we do not encourage people to change it, it is only beneficial for server hosters to allow multiple instances run in parallel.

**Comment by IntelOrca** [comment]
Or portable / USB flash drive installs.

**Comment by wisepotato** [comment]
> You can already save/load your parks from wherever you like. Also you can already override the data directory via argument but we do not encourage people to change it, it is only beneficial for server hosters to allow multiple instances run in parallel.

So question, why encourage users not to change it? Is it technical? What really could go wrong?

**Comment by thewesker** [comment]
> Or portable / USB flash drive installs.

I specifically like this as my install is placed in OneDrive so it is always in sync no matter which PC I use.

**Comment by janisozaur** [comment]
I don't think we ever said we discourage it, the option is there properly working and documented, so you're just fine to set it to whatever you want. The issue is specifically about the defaults.

**Comment by ZehMatt** [comment]
> > You can already save/load your parks from wherever you like. Also you can already override the data directory via argument but we do not encourage people to change it, it is only beneficial for server hosters to allow multiple instances run in parallel.
> 
> So question, why encourage users not to change it? Is it technical? What really could go wrong?

Less configurations means less can go wrong which applies for mostly all software.

**Comment by zedd-0** [comment]
it is 2023 and the game still put the data to "Documents\OpenRCT2"
this is my Office-folder, only for MY Documents and not a GAME-DUMP.

please change this.

by the way:
your game cannot store anything to "Documents\", because i restricted my folder with windows defender.
so you >>have to<< change it. for security reasons.

sorry.

thx 4 reading
have a nice day.

**Comment by 733737** [comment]
zedd-0, maybe you could buy a 2nd computer for personal use, if you are having trouble with openrct2 on a work laptop?

**Comment by IntelOrca** [comment]
@zedd-0 OpenRCT2 has a portable mode, you can pass it command line arguments to use a custom location for user data.

**Comment by janisozaur** [comment]
@zedd-0 pass any path you desire with `--user-data-path`

**Comment by zedd-0** [comment]
- buying a 2end pc is not an option for me, sry. and do not solve the problem.

- i use already the "portable mode" with the '-user-data-path' -command.
but i do not understand, why is the potable package not already in the potable mode.
other apps i use, use something like a 'setting.ini' to mark there app as portable.

it would be nice, if the installed version use windows %APPDATA% directory instant of my documents directory.
I don't even know what variable is used for the documents folder...

**Comment by trzejos** [comment]
Personally, I am fine with the current setup. I like the idea of having everything in one place so that backups are easier. I wouldn't mind it being moved out of Documents, but I do not want it split up so I don't have to worry about multiple directories.

**Comment by IntelOrca** [comment]
@zedd-0 historical reasons and based on what OpenTTD did. We cannot change the way it is done as it will break a lot of the current users. OpenLoco uses %APPDATA%.

If the portable mode does work for you then we will close the issue as wontfix.

**Comment by madcampos** [comment]
Could this be configured and by default point to the documents folder so it won't break people setups and will allow for the folder to be changed?
